vaadin-radio-button-flow-integration-tests/src/main/java/com/vaadin/flow/component/radiobutton/tests/RadioButtonGroupDataViewPage.java

```diff
@@ -0,0 +1,194 @@
+import java.util.stream.Stream;
+
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.NativeButton;
+import com.vaadin.flow.component.radiobutton.RadioButtonGroup;
+import com.vaadin.flow.component.radiobutton.dataview.RadioButtonGroupDataView;
+import com.vaadin.flow.component.radiobutton.dataview.RadioButtonGroupListDataView;
+import com.vaadin.flow.data.provider.AbstractDataProvider;
+import com.vaadin.flow.data.provider.Query;
+import com.vaadin.flow.router.Route;
+
+@Route("radio-Button-group-data-view")
+public class RadioButtonGroupDataViewPage extends Div {
+
+    private static final String FIRST = "first";
+    private static final String SECOND = "second";
+    private static final String CHANGED_1 = "changed-1";
+    private static final String CHANGED_2 = "changed-2";
+
+    public RadioButtonGroupDataViewPage() {
+        createGenericDataViewRadioButtonGroup();
+        createListDataViewRadioButtonGroup();
+        createAddItemByDataViewRadioButtonGroup();
+        createFilterItemsByDataViewRadioButtonGroup();
+    }
+
+    private void createGenericDataViewRadioButtonGroup() {
+        Item first = new Item(1L, FIRST);
+        Item second = new Item(2L, SECOND);
+
+        List<Item> items = new ArrayList<>(Arrays.asList(first, second));
+        GenericDataProvider dataProvider = new GenericDataProvider(items);
+
+        RadioButtonGroup<Item> rgbForDataView = new RadioButtonGroup<>();
+        rgbForDataView.setId("rgbForDataView");
+
+        RadioButtonGroupDataView<Item> dataView = rgbForDataView
+                .setItems(dataProvider);
+        dataView.setIdentifierProvider(Item::getId);
+
+        NativeButton dataViewUpdateButton = new NativeButton("Update",
+                click -> {
+                    first.setValue(CHANGED_1);
+                    second.setValue(CHANGED_2);
+
+                    dataView.refreshItem(new Item(1L));
+                });
+        dataViewUpdateButton.setId("updBtnGdpDv");
+
+        add(rgbForDataView, dataViewUpdateButton);
+    }
+
+    private void createListDataViewRadioButtonGroup() {
+        Item first = new Item(1L, FIRST);
+        Item second = new Item(2L, SECOND);
+
+        RadioButtonGroup<Item> rgbForListDataView = new RadioButtonGroup<>();
+        rgbForListDataView.setId("rgbForListDataView");
+
+        RadioButtonGroupListDataView<Item> dataView = rgbForListDataView
+                .setItems(first, second);
+        dataView.setIdentifierProvider(Item::getId);
+
+        NativeButton dataViewUpdateButton = new NativeButton("Update",
+                click -> {
+                    first.setValue(CHANGED_1);
+                    second.setValue(CHANGED_2);
+
+                    dataView.refreshItem(new Item(1L));
+                });
+        dataViewUpdateButton.setId("updBtnLstDv");
+
+        add(rgbForListDataView, dataViewUpdateButton);
+    }
+
+    private void createAddItemByDataViewRadioButtonGroup() {
```

Comment:
Why are you highlighting only the 'addItem' method for being IT tested? Basically, `AbstractListDataView` is tested in Flow and RBG list data view doesn't bring any extra implementation, so for me it feels like either not test it here, or test all the in-memory API.

Reply:
I haven't seen this from your point of view. I just wanted to have test-cases on different `setItems` methods. One with providing the `DataProvider` and another one using bare items being passed to the `setItems` method.

Reply:
I didn't get the part you mentioned about "**only the addItem method is being IT tested**". I understand that `RadioButtonGroupListDataView` brings no more implementation, so no need to test it, but since I added the DataView features, I thought having ITs to show the whole thing works with RBG would be somehow necessary. 

Currently, tests are covering `refreshItem`, `addItem`, `setFilter` and `removeFilter` . Do you recommend to add one more IT to cover `removeItem` or maybe other methods? Or just wipe all the ITs on `RadioButtonGroupListDataView`?

Reply:
I mean, `refreshItem`, `addItem`, `setFilter` and `removeFilter` are just few of those 20 methods in AbstractListDataView, so why do we test only them and skip others? Oh, it's a generic data view test! Ok, maybe it's fine to keep only limited number of tests, just to smoke test the data view. Alright, let's then create a comment explaining it in RadioButtonGroupDataViewPage (that those tests some kind of smoke tests to cover the whole behaviour).